test/unit/SourceLocator/Type/DirectorySourceLocatorTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace BetterReflectionTest\SourceLocator\Type;
+
+use BetterReflection\Reflector\ClassReflector;
+use BetterReflection\SourceLocator\Type\DirectorySourceLocator;
+
+/**
+ * @covers \BetterReflection\SourceLocator\Type\DirectorySourceLocator
+ */
+class DirectorySourceLocatorTest extends \PHPUnit_Framework_TestCase
+{
+
+    private $directoryToScan = __DIR__ . '/../../Assets/DirectoryScannerAssets';
+
+    /**
+     * @var DirectorySourceLocator
+     */
+    private $sourceLocator;
+
+    public function setUp()
+    {
+        $this->sourceLocator = new DirectorySourceLocator([$this->directoryToScan]);
+    }
+
+    public function testScanDirectoryClasses()
+    {
+        $reflector = new ClassReflector($this->sourceLocator);
+        $this->assertCount(2, $reflector->getAllClasses());
```

Comment:
Can you assert on the class names being retrieved? You can use `sort()` in the test to ensure they are sorted by FQCN